components/slack/slack.app.js

```diff
@@ -3,6 +3,243 @@ const { WebClient } = require('@slack/web-api')
 module.exports = {
   type: "app",
   app: "slack",
+  propDefinitions: {
+    publicChannel: {
+      type: "string",
+      label: "Channel",
+      //description: "Select one or more channels or DM conversations to monitor for new messages.",
+      async options({ prevContext }) {
+        let { types, cursor, userNames } = prevContext
+        if (types == null) {
+          const scopes = await this.scopes()
+          types = ["public_channel"]
+        }
+        const resp = await this.availableConversations(types.join(), cursor)
+        return {
+          options: resp.conversations.map((c) => {
+            if (c.is_im) {
+              return { label: `Direct messaging with: @${userNames[c.user]}`, value: c.id }
+            } else if (c.is_mpim) {
+              return { label: c.purpose.value, value: c.id }
+            } else {
+              return { label: `${c.name}`, value: c.id }
+            }
+          }),
+          context: { types, cursor: resp.cursor, userNames },
+        }
+      },
+    },
+    privateChannel: {
+      type: "string",
+      label: "Channel",
+      //description: "Select one or more channels or DM conversations to monitor for new messages.",
+      async options({ prevContext }) {
+        let { types, cursor, userNames } = prevContext
+        if (types == null) {
+          const scopes = await this.scopes()
+          types = ["private_channel"]
+        }
+        const resp = await this.availableConversations(types.join(), cursor)
+        return {
+          options: resp.conversations.map((c) => {
+            return { label: `${c.name}`, value: c.id }
+          }),
+          context: { types, cursor: resp.cursor, userNames },
+        }
+      },
+    },
+    user: {
+      type: "string",
+      label: "User",
+      //description: "Select one or more channels or DM conversations to monitor for new messages.",
+      async options({ prevContext }) {
+        let { types, cursor, userNames } = prevContext
+        if (types == null) {
+          const scopes = await this.scopes()
+          types = ["im"]
+          // TODO use paging
+          userNames = {}
+          for (const user of (await this.users()).users) {
```

Comment:
I think this will cause a huge delay in async options for slack with many users, and I'm not sure that this is useful at all.  The other comments lay out some alternatives.